### DISCUSSION TITLE
Seeking Guidance on Using Istio for Secure External Access and mTLS in Private Kubernetes Cluster

### DISCUSSION THREAD

**Discussion by nahian252** [istio/istio#52676] [Q&A]
Hi Istio team,

I have a private Kubernetes cluster with several pods running, and I’m looking to provide access to a specific pod for users who are external to my network. I want to ensure that this access is secured using mutual TLS (mTLS) to adhere to a Zero Trust approach.

Questions:

Can Istio help in implementing mTLS for external access to a specific pod?
What are the best practices for configuring Istio to achieve this setup?
How does Istio handle the integration of external users and certificates?

Any guidance or recommendations on how to achieve this with Istio would be greatly appreciated!

Thank you!

**Comment by howardjohn** [comment]
Typically to expose to an external client you are doing this with a gateway (https://istio.io/latest/docs/tasks/traffic-management/ingress/ingress-control/). You can either terminate their mTLS and then initiate a new mTLS connection to the pod, or use the passthrough tls mode to just proxy the original stream end to end.

The big challenge is really what certificates are your clients using and how do they get them?
